platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java

```diff
@@ -231,7 +234,8 @@ private AddressBook checkCandidateAddressBookValidity(@Nullable final AddressBoo
      *
      * @param usedAddressBook the address book to be returned from the AddressBookInitializer.
      */
-    private void recordAddressBooks(@NonNull final AddressBook usedAddressBook) {
+    private synchronized void recordAddressBooks(@NonNull final AddressBook usedAddressBook) {
+        cleanAddressBookDirectory();
```

Comment:
Should cleanup happen *after* the write? (probably doesn't matter too much either way)

Reply:
Whether it happens before or after, the number of file created on disk is always MAX + 2.    I will move it after so that the final number of files is consistent with the specified max.